exporter/cassandra/exporter.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/jaegertracing/jaeger/pkg/cassandra"
+	casConfig "github.com/jaegertracing/jaeger/pkg/cassandra/config"
+	plugin "github.com/jaegertracing/jaeger/plugin/storage/cassandra"
+	cSpanStore "github.com/jaegertracing/jaeger/plugin/storage/cassandra/spanstore"
+	"github.com/jaegertracing/jaeger/plugin/storage/cassandra/spanstore/dbmodel"
+)
+
+func newTracesExporter(cfg *Config, set component.ExporterCreateSettings) (component.TracesExporter, error) {
+	s := newSender(cfg, set.TelemetrySettings)
+	return newTracesExporterWithSender(cfg, set, s)
+}
+
+func newTracesExporterWithSender(cfg *Config, set component.ExporterCreateSettings, sender *sender) (component.TracesExporter, error) {
+	return exporterhelper.NewTracesExporter(
+		context.TODO(), set, cfg, sender.pushTraces,
+		exporterhelper.WithCapabilities(consumer.Capabilities{MutatesData: false}),
+		exporterhelper.WithStart(sender.start),
+		exporterhelper.WithShutdown(sender.shutdown),
+		exporterhelper.WithTimeout(cfg.TimeoutSettings),
+		exporterhelper.WithRetry(cfg.RetrySettings),
+		exporterhelper.WithQueue(cfg.QueueSettings),
+	)
+}
+
+type sender struct {
+	name           string
+	settings       component.TelemetrySettings
+	cfg            *Config
+	primaryConfig  casConfig.SessionBuilder
+	primarySession cassandra.Session
+	// factory  *plugin.Factory // Cannot use unless some functions or fields are exported
+	writer *cSpanStore.SpanWriter
+}
+
+func newSender(cfg *Config, settings component.TelemetrySettings) *sender {
+	return &sender{
+		name:          cfg.ID().String(),
+		settings:      settings,
+		cfg:           cfg,
+		primaryConfig: cfg.Options.GetPrimary(),
+	}
+}
+
+func (s *sender) pushTraces(ctx context.Context, td ptrace.Traces) error {
+	batches, err := jaeger.ProtoFromTraces(td)
+	if err != nil {
+		return consumererror.NewPermanent(fmt.Errorf("failed to push trace data via cassandra exporter: %w", err))
+	}
+
+	for _, batch := range batches {
+		process := batch.GetProcess()
+		if process == nil {
+			s.settings.Logger.Debug("NO PROCESS")
+		}
+		for _, span := range batch.GetSpans() {
+			span.Process = process
+			err = s.writer.WriteSpan(ctx, span)
+			if err != nil {
+				s.settings.Logger.Debug("failed to push trace data to cassandra", zap.Error(err))
+				return fmt.Errorf("failed to push trace data via cassandra exporter: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (s *sender) start(_ context.Context, host component.Host) error {
+	primarySession, err := s.primaryConfig.NewSession(s.settings.Logger)
+	if err != nil {
+		return err
+	}
+	s.primarySession = primarySession
+
+	options, err := writerOptions(s.cfg.Options)
+	if err != nil {
+		return err
+	}
+
+	s.writer = cSpanStore.NewSpanWriter(
+		primarySession,
+		s.cfg.Options.SpanStoreWriteCacheTTL,
+		metrics.New("exporter_cassandra"),
+		s.settings.Logger,
+		options...,
+	)
+
+	/* We can't use the factory due to the way servers is populated in GetPrimary
```

Comment:
This line would override whatever is defined in the configuration: 	

https://github.com/jaegertracing/jaeger/blob/cac21f82950545b613f45d3984d96e1498bfd15d/plugin/storage/cassandra/options.go#L268

The unexported `servers` , `127.0.0.1` by default, overrides the `Servers` field in the configuration. It's then added to the unexported `primaryConfig` in the factory which I can't get at either. Maybe a simple setter in the factory would let me have a clean hookup here. I don't believe we would care about the archive at all in an otel world. It would be a secondary cassandra exporter in the otel config.